public/app/plugins/datasource/influxdb/components/VisualInfluxQLEditor/Editor.test.tsx

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+import { InfluxQuery } from '../../types';
+import InfluxDatasource from '../../datasource';
+import { render } from '@testing-library/react';
+import { Editor } from './Editor';
+
+// we mock the @grafana/ui components we use to make sure they just show their "value".
```

Comment:
This is cool 😎 I like how easily it is too assert the state of entire editor with a string. Maybe we could do it for other data sources?

Reply:
thanks 😁 sure, feel free to do it somewhere else too. though i wonder that maybe there is a less "homemade" approach to this. (maybe some "rendering-mode-thing" for grafana-ui, or a premade mock-all-of-them function as part of grafana-ui...not sure)

Reply:
Is this basically a snapshot test for the whole query editor?

Although I'm not a big fan of snapshot tests, I can see why you would do something like this. I'm just wondering if mocking `@grafana/ui` is a reliable approach, as you will basically hide any change there and this test will "always" pass. basically at this point this test will have its own life, independently of the used components and if they work or not.

Did you consider writing a small snapshotSerializer for jest instead? you would have the same benefits but by using the actual components the user will see on the screen.

Reply:
"Whole query editor" => "text representation of the query being shown"

Reply:
@Elfo404 jest snapshotSerializer ... interesting.. i never tried that, maybe i could use those. i looked at using snapshots, but they seemed to store the html-structure, so i did not go there. maybe with a custom serializer i could do what i need... i agree that snapshots are usually not a good plan, i just wanted something at the "highest" level, a small check to kind of make sure the parts of the form are at the correct places etc.

Reply:
maybe it's a bit of an overkill though. not really sure.

Reply:
I also like how the tests read in this format, but also wonder if we could use the grafana/ui components and just serialize them in a similar way, which I think would catch some integration bugs with grafana/ui which are always those which are harder to catch.

Reply:
@aocenas "serialize them in a similar way" you mean @Elfo404 's approach or some other way?